### DISCUSSION TITLE
Worse Results on Newer Versions

### DISCUSSION THREAD

**Discussion by sisenior** [d8ahazard/sd_dreambooth_extension#579] [Q&A]
Hello, I've been using this extension since last month and everything worked fine until one day it didn't work anymore.

Models trained on newer versions (since the filewords update I think) have a ton of bleeding issues and prompts doesn't work well too. Here's a comparison (both trained using the same dataset and same class images):

**NEWER VERSIONS**

![new](https://user-images.githubusercontent.com/120907029/208288546-94b76ef7-c7dd-4415-8d87-8a7d8401a686.png)

---------------------------------------------------------------------------------------------------------------------------------------------------
**OLD VERSION**

![old](https://user-images.githubusercontent.com/120907029/208288563-1ac9ab4c-cd62-4a75-b312-ac5d6fa7a791.png)

I found that the diffusers version where I'm not getting bleeding issues is a "custom" version of diffusers 0.8.0.dev0 from Nov 9 (is the diffusers version that the old extension was using before Nov 12)
[674823c](https://github.com/huggingface/diffusers/commit/674823c8e4582868e7f30682868dcd216c9972f8)

I already added that version of diffussers in the requirements.txt file, Also in the requirements.txt files from web-ui I changed the transformers version to `transformers==4.19.2`, this is important because other transformers versions gives me OOM errors even with `--xformers`

Dreamboot:
[requirements.txt](https://github.com/d8ahazard/sd_dreambooth_extension/files/10326589/requirements.txt)

Web-UI
[requirements_versions.txt](https://github.com/d8ahazard/sd_dreambooth_extension/files/10326592/requirements_versions.txt)
[requirements.txt](https://github.com/d8ahazard/sd_dreambooth_extension/files/10326610/requirements.txt)

I'm using this Dreambooth extension commit [65a4bdf](https://github.com/d8ahazard/sd_dreambooth_extension/commit/65a4bdffc41b3ab44ec8262acce21c87d9a7fccf)

And I'm using this Web-UI commit [7ba3923](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/7ba3923d5b494b7756d0b12f33acb3716d830b9a)

With those commits and requirements.txt files I don't have the bleeding issue.

I have a 3060 (12gb vram) and I use `--xformers`
I already tried everything on the newer versions of the extension, using [filewords] with my images already captioned, using `flash_attention` or `xformers`, different learning rates, different tokens, different training steps and nothing, all trained models have the same bleeding issue (or overfitting) no matter what I do.

Please I need help, I don't want to be stuck forever on that ancient commit.

**Comment by nanafy** [comment]
I actually made the exact same posts as you!
https://github.com/d8ahazard/sd_dreambooth_extension/discussions/577

Thanks for sharing, now I know I am not the only one!! Which means it can be fixed yay!

**Comment by ArrowM** [comment]
I created issue #605, but I don't have a lot of info. Please add details to the issue thread if you have any.

We have a few people trying to investigate in the extension's discord. We are compainge tons of models generated across different versions  of the extension and with different dreambooth input params. Hop in if you wanna help: https://discord.com/channels/1049560328654037023/1055692394168856577

**Reply by sisenior** [reply]
I don't think I have much information to share. But I can provide the requirements.txt file where I don't get the bleeding issue.

I found that the diffusers version where I'm not getting bleeding issues is a "custom" version of diffusers 0.8.0.dev0 from Nov 9 (is the version that the old extension was using before Nov 12)
[674823c](https://github.com/huggingface/diffusers/commit/674823c8e4582868e7f30682868dcd216c9972f8)

I already added that version of diffussers in the requirements.txt file 

Dreamboot:
[requirements.txt](https://github.com/d8ahazard/sd_dreambooth_extension/files/10326589/requirements.txt)

Web-UI
[requirements_versions.txt](https://github.com/d8ahazard/sd_dreambooth_extension/files/10326592/requirements_versions.txt)
[requirements.txt](https://github.com/d8ahazard/sd_dreambooth_extension/files/10326610/requirements.txt)


I'm using this Dreambooth extension commit [65a4bdf](https://github.com/d8ahazard/sd_dreambooth_extension/commit/65a4bdffc41b3ab44ec8262acce21c87d9a7fccf)

And I'm using this Web-UI commit [7ba3923](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/7ba3923d5b494b7756d0b12f33acb3716d830b9a)

I hope this can be helpful

**Reply by mykeehu** [reply]
I'm a couple of versions ahead of the DB, where only the interface has been reworked, not the engine, so I'm working on that with your modified requirements.txt files without any problems:
[4684f0faea19c243216eac2278a42fcfcbbd756b](https://github.com/d8ahazard/sd_dreambooth_extension/tree/4684f0faea19c243216eac2278a42fcfcbbd756b)

There is now also .ckpt generation and a preview image.

**Reply by sisenior** [reply]
Thanks, I will try with that version. Also in the requirements.txt files from web-ui I changed the transformers version to `transformers==4.19.2`, this is important because other transformers versions gives me OOM errors even with `--xformers`

**Reply by mykeehu** [reply]
It runs perfectly for me, because this DB hasn't used xformers yet, and what's interesting is that I can train with it on batch 10 with a 3060 card with 12 GB.

**Comment by Sovrer** [comment]
DUDE you saved my models, installed the commits and the txt and it works fantastic. And this old version doesn't have that filewords stuff and doesn't have the hundreds of options that not even the devs known what they do nor even care to explain. Just simple straightforward training without unnecessary chores.
